src/examples/libpmem2/pmemset/pmemset.h

```diff
@@ -0,0 +1,345 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2020, Intel Corporation */
+
+#ifndef LIBPMEMSET_H
+#define LIBPMEMSET_H 1
+
+#include <stddef.h>
+#include <stdint.h>
+#include <sys/types.h>
+#include <libpmem2.h>
+
+struct pmemset;
+struct pmemset_config;
+struct pmemset_source;
+struct pmemset_part_descriptor;
+struct pmemset_part;
+
+int pmemset_config_new(struct pmemset_config **config);
+
+/*
+ * By default, the library only attempts to open the parts. These two functions
+ * modify this behavior.
+ * Create if none behaves like O_CREAT, but also initializes part metadata.
+ * Create if invalid checks if the existing file contains a valid pool,
+ * and if not, creates a new one (this is useful for close-to-open).
+ */
+void pmemset_config_set_create_if_none(struct pmemset_config *config,
```

Comment:
What about  one function **pmemset_config_set_create** with flag or optional config parameter.

Reply:
Because we moved the header out of the parts, `create_if_invalid` is no longer needed - this is something that the user will need to handle manually.